lib/mm_input.rb

```diff
@@ -0,0 +1,78 @@
+HELP = 'HELP'
+QUIT = 'QUIT'
+EXIT = 'EXIT'
+RULES = 'RULES'
+
+class Input
+    attr_reader :user_input, :is_valid, :output_code, :restart_request, :show_help, :show_rules, :err_data
+
+    def initialize
+        @show_help = false
+        @show_rules = false
+        puts 'Please enter a code or command...'
+        print "---> "
+        validate(gets.chomp)
+    end
+
+    def validate(to_validate)
+        @err_data = {
+            err: false,
+            bad_chars: [],
+            too_long: [],
+            too_short: []
+        }
```

Comment:
This is a smell to me. What if your game expands to allow for more errors? This list could get really long really fast. Can we think of a more dynamic way of capturing errors? Can we just have one `errors` array?